Jaybee84/move ensemble (#103)
jaybee84/ml-in-rd: content/04.model-complexity.md

A variation of elastic-net, where a two-step regularized logistic regression was used to pre-select an optimal number of genes before implementing elastic-net regularization for gene selection, identified immune cell signatures in an RNA-seq dataset where the number of cells sampled were far fewer than number of genes profiled [@doi: 10.1186/s12859-019-2994-z].  
Thus, regularization methods like LASSO or elastic-net have been methods of choice where the profiled feature space is substantially larger than the number of samples; these methods should be explored while working with rare disease datasets.

#### Ensemble learning

> Rare disease datasets not only present few data points but the data points also come with high label-uncertainty thus forming silver standard datasets.

Reviewer: It _very well_ may have been my suggestion to kick the gold standard vs. silver standard discussion to the introduction but seeing this makes me think there's a non-zero chance that if someone is not familiar with this concept prior to reading the paper, they will have forgotten what it means by the time they make it to this section.

Author: ```suggestion
Rare disease datasets not only present limited data points but, due to inadequate understanding of the diseases, the labels associated with the data points also carry high levels of uncertainty. 
Training ML models using data points with high label-uncertainty (i.e. a silver standard dataset) can lead to unstable predictions.
```

Author: rephrased this

> Implementing machine learning on such data can lead to unstable predictions. 

Author: ```suggestion
```

In such cases, combining various machine learning methods together (_ensemble learning_) can increase accuracy and stability of the predictions.

> For example, ensemble learning methods like random forests use bagging of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features hidden in the dataset [@doi:10.1186/1472-6947-13-134, @doi:10.1023/A:1010933404324, @doi:10.1214/aos/1031689014, @doi:10.1177/2045894019890549, @doi:10/btzfh6].

Reviewer: ```suggestion
For example, ensemble learning methods like random forests use bagging of independent decision trees that use similar parameters but different paths to form a consensus about the important predictive features hidden in the dataset [@doi:10.1186/1472-6947-13-134; @doi:10.1023/A:1010933404324; @doi:10.1214/aos/1031689014; @doi:10.1177/2045894019890549; @doi:10/btzfh6]. 
```

I believe.

> In silver standard datasets, the limited success of the bagging approach has led to the use of ensemble learning or _cascade learning_, where multiple methods leveraging distinct underlying assumptions are used in tandem and augmented with algorithms like AdaBoost (_boosting_), to capture stable patterns existing in the silver standard data and reduce uncertainty [@doi:10.1109/CVPR.2001.990537, @doi:10.1007/978-3-540-75175-5_16, @doi:10.1109/ICPR.2004.1334680].

Reviewer: ```suggestion
In silver standard datasets, the limited success of the bagging approach has led to the use of ensemble learning or _cascade learning_, where multiple methods leveraging distinct underlying assumptions are used in tandem and augmented with algorithms like AdaBoost (_boosting_), to capture stable patterns existing in the silver standard data and reduce uncertainty [@doi:10.1109/CVPR.2001.990537; @doi:10.1007/978-3-540-75175-5_16; @doi:10.1109/ICPR.2004.1334680].
```
